asab/library/providers/git.py

```diff
@@ -134,15 +136,33 @@ def fetch(self):
 
 
 	def pull(self):
-
 		new_commit_id = self.fetch()
 
+		# Before new head is set, check the diffs. If changes in subscribed directory occured, set `publish` flag.
+
+		publish = False
+		for path in self.SubscribedPath:
+			for i in self.GitRepository.diff(self.GitRepository.head.target, new_commit_id).deltas:
+				if path == "/":
+					publish = True
+				elif ("/" + i.old_file.path).startswith(path):
+					publish = True
+			if publish is True:
+				break
+
 		if new_commit_id == self.GitRepository.head.target:
 			return
 
+		# Reset HEAD
 		self.GitRepository.head.set_target(new_commit_id)
 		self.GitRepository.reset(new_commit_id, pygit2.GIT_RESET_HARD)
-		self.App.PubSub.publish("GitProviderUpdated!", self)
+
+		# Once reset of the head is finished, PubSub message about the change in the subsrcibed directory gets published.
+		if publish:
+			self.App.PubSub.publish("ASABLibrary.change!", self)
```

Comment:
`self.App.PubSub.publish("ASABLibrary.change!", self, paths=pubchanges)`

Reply:
I'm not publishing exact paths that have changed, but paths that the "user" has subscribed to and in which something happened, no matter what.